splunklogger.js

```diff
@@ -107,6 +136,58 @@ var defaultRequestOptions = {
 };
 
 /**
+ * Disables the interval timer set by <code>this._enableTimer()</code>.
+ *
+ * param {Number} interval - The batch interval.
+ * @private
+ */
+SplunkLogger.prototype._disableTimer = function() {
+    if (this._timerID) {
+        clearInterval(this._timerID);
+        this._timerDuration = 0;
+        this._timerID = null;
+    }
+};
+
+/**
+ * Configures an interval timer to flush any events in
+ * <code>this.contextQueue</code> at the specified interval.
+ *
+ * param {Number} interval - The batch interval.
+ * @private
+ */
+SplunkLogger.prototype._enableTimer = function(interval) {
+    // Only enable the timer if possible
+    if (typeof interval === "number") {
```

Comment:
Up to you, but I'd make a set of validation functions for numbers, positive numbers, etc.